node/src/components/consensus/era_supervisor/tests.rs

```diff
@@ -0,0 +1,125 @@
+use anyhow::Error;
+use casper_execution_engine::{core::engine_state::genesis::GenesisAccount, shared::motes::Motes};
+use derive_more::From;
+use prometheus::Registry;
+
+use super::*;
+use crate::{
+    components::{
+        consensus::{
+            tests::mock_proto::{self, MockProto, NodeId},
+            Config,
+        },
+        storage::Storage,
+        Component,
+    },
+    crypto::asymmetric_key::{PublicKey, SecretKey},
+    effect::{
+        announcements::ConsensusAnnouncement,
+        requests::{
+            BlockExecutorRequest, BlockProposerRequest, BlockValidationRequest,
+            ContractRuntimeRequest, NetworkRequest, StorageRequest,
+        },
+        EffectBuilder,
+    },
+    protocol,
+    reactor::{EventQueueHandle, QueueKind, Scheduler},
+    testing::TestRng,
+    utils::{self, External, Loadable},
+};
+
+type ClMessage = mock_proto::Message<ClContext>;
+
+#[derive(Debug, From)]
+enum Event {
+    #[from]
+    Consensus(super::Event<NodeId>),
+    #[from]
+    Network(NetworkRequest<NodeId, protocol::Message>),
+    #[from]
+    BlockProposer(BlockProposerRequest),
+    #[from]
+    ConsensusAnnouncement(ConsensusAnnouncement),
+    #[from]
+    BlockExecutor(BlockExecutorRequest),
+    #[from]
+    BlockValidator(BlockValidationRequest<ProtoBlock, NodeId>),
+    #[from]
+    Storage(StorageRequest<Storage>),
+    #[from]
+    ContractRuntime(ContractRuntimeRequest),
+}
+
+/// Loads the local chainspec and overrides timestamp and genesis account with the given stakes.
+fn new_test_chainspec(stakes: Vec<(PublicKey, u64)>) -> Chainspec {
+    let mut chainspec = Chainspec::from_resources("local/chainspec.toml");
+    chainspec.genesis.accounts = stakes
+        .into_iter()
+        .map(|(pk, stake)| {
+            let motes = Motes::new(stake.into());
+            GenesisAccount::new(pk.into(), pk.to_account_hash(), motes, motes)
+        })
+        .collect();
+    chainspec.genesis.timestamp = Timestamp::now() + 1000.into();
+    chainspec.genesis.highway_config.genesis_era_start_timestamp = chainspec.genesis.timestamp;
+    chainspec
+}
```

Comment:
Obviously, not in this PR but I am not a fan of passing in _whole_ Chainspec. Especially that it has four fields that are Wasm contracts in byte array form. Can we ticket a refactor for it? So that only the necessary subset of data is passed in to the constructor of `EraSupervisor`?

Reply:
Done. [NDRS-578](https://casperlabs.atlassian.net/browse/NDRS-578)